core/service.go

```diff
@@ -36,15 +36,17 @@ type Service struct {
 	rt *runtime.Runtime
 	b  *babe.Session
 
-	msgChan <-chan []byte
+	recChan  <-chan []byte
+	sendChan chan<- []byte
 }
 
 // NewService returns a Service that connects the runtime, BABE, and the p2p messages.
-func NewService(rt *runtime.Runtime, b *babe.Session, msgChan <-chan []byte) *Service {
+func NewService(rt *runtime.Runtime, b *babe.Session, msgChan <-chan []byte, sendChan chan<- []byte) *Service {
```

Comment:
I think this might break the cmd tests since in cmd/gossamer/configcmd.go when we call `makeNode` there isn't a sendChan passed in. we need to make sure the same channels get passed into core and p2p (as well as babe now)

also, there might be an issue with sendChan being read-only, I think we should send messages to be sent to p2p into the sendChan. otherwise, if babe puts the BlockAnnounce messages into the receive-only channel, we won't be able to tell whether they can from p2p or babe.